src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -324,6 +324,15 @@ private async Task<ReindexJobQueryStatus> ProcessQueryAsync(ReindexJobQueryStatu
                     {
                         _reindexJobRecord.Progress += results.Results.Count();
                         query.Status = OperationStatus.Completed;
+
+                        // Remove oldest completed queryStatus object if count > 10
+                        // to ensure reindex job document doesn't grow too large
+                        if (_reindexJobRecord.QueryList.Keys.Where(q => q.Status == OperationStatus.Completed).Count() > 10)
```

Comment:
>_reindexJobRecord.QueryList.Keys.Where(q => q.Status == OperationStatus.Completed [](start = 28, length = 81)

I can be wrong but it looks like there is 1:1 relationship between  QueryList and Resources, where QueryList has some extra fields.
We want it clear, because Continuation token is big, and whole collection is taking too much space? I just find it's interesting to clear QueryList, but leave Resources as is.

Reply:
Good question!  It is actually a 1:many relationship between Resource:QueryList.  If an initial query for a given resource has a continuation token, we will add a new entry for that continuation token, and then continue until the query no longer returns a continuationToken.

Reply:
Thanks for clarification.
Another silly question, why can't we just remove `query` object from `QueryList`, why wait for 10 of them and them remove oldest one.

---
In reply to: [587903115](https://github.com/microsoft/fhir-server/pull/1701#discussion_r587903115) [](ancestors = 587903115)

Reply:
I had a thought that if something did go wrong during the reindex job, it might be useful to see the last few queries that completed successfully.